.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -4501,14 +4501,14 @@ stages:
           GITHUB_TOKEN: $(GITHUB_TOKEN)
 
 - stage: throughput
-  condition: and(succeeded(), eq(variables.isMainRepository, true))
+  # By default throughput tests happen on release or master branches only, or when running a benchmarks only build. Overridable by setting 'run_extended_throughput_tests' to true
+  condition: and(succeeded(), eq(variables.isMainRepository, true), or(eq(variables.isMainOrReleaseBranch, 'true'), not(eq(variables['isBenchmarksOnlyBuild'], 'False')), eq(variables.run_extended_throughput_tests, 'true')))
```

Comment:
NIT: Just a curiosity... Why do we use "False" and "true"?

Reply:
That's a good question 🤔 IIRC it's to do with how Azure Devops handles booleans. The `isBenchmarksOnlyBuild` variable is calculated, and so it essentially does a `.ToString()` on the result, giving `True` and `False`. But honestly, I'm not sure if that's right, because then why is `isMainOrReleaseBranch` checking against `true`?! 🤷‍♂️ 😅